pkg/apis/componentconfig/types.go

```diff
@@ -62,3 +62,213 @@ const (
 	ProxyModeUserspace ProxyMode = "userspace"
 	ProxyModeIPTables  ProxyMode = "iptables"
 )
+
+type KubeletConfiguration struct {
```

Comment:
The convention is that, but the actual behavior for certain fields is that omit empty *obscures* the user's actual values (for []string and map) because it replaces nil with empty map.  Even though I prefer writing all the fields out for user config (so a user can figure out what config they *can* set), we should never set arrays or maps to omit empty.